File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0;
        int r = height.size()-1;
        long ans = 0;
        while(l < r) {
            ans = max(ans, (long)((r-l)*min(height[l], height[r])));

            if (height[l] < height[r]) l++ ;
            else r--;
        }

        return ans;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int i = 0;
        int j = height.length-1;
        int ans = 0;

        while(i < j) {
            ans = Math.max(ans, Math.min(height[i], height[j]) * (j-i));
            if (height[i] >= height[j]) {
                j--;
            } else {
                i++;
            }
        }

        return ans;
    }
}